pkg/api/vlabs/validate.go

```diff
@@ -264,6 +274,17 @@ func validateUniqueProfileNames(profiles []AgentPoolProfile) error {
 	return nil
 }
 
+func validateClassicProfileType(profileType ClassicAgentPoolProfileType) error {
+	switch profileType {
+	case SwarmPublic:
+	case DCOSPublic:
+	case DCOSPrivate:
+	default:
+		return fmt.Errorf("ClassicAgentPoolProfile has unknown profile type: %s", profileType)
```

Comment:
can you just add case "", since case "" is a valid value meaning not classic.  It may be useful to declare a const for "" and call it "NotClassic" or similar name